### PR TITLE
[refactor](Nereids): refactor PushdownLimit

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem summary

refactor PushdownLimit

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.39 seconds
 stream load tsv:          478 seconds loaded 74807831229 Bytes, about 149 MB/s
 stream load json:         40 seconds loaded 2358488459 Bytes, about 56 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230302152047_clickbench_pr_107463.html

**Comment by jackwener** [comment]
run buildall

**Comment by jackwener** [comment]
run buildp0

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java [hunk 1]
```diff
@@ -67,65 +74,45 @@ public List<Rule> buildRules() {
...
+                                return null;
+                            }
                             return limit.withChildren(union.withChildren(newUnionChildren));
                         })
-                        .toRule(RuleType.PUSH_LIMIT_THROUGH_UNION)
+                        .toRule(RuleType.PUSH_LIMIT_THROUGH_UNION),
+                logicalLimit(logicalOneRowRelation())
+                        .then(limit -> limit.getLimit() > 0
+                                ? limit.child() : new LogicalEmptyRelation(limit.child().getOutput()))
+                        .toRule(RuleType.PUSH_LIMIT_THROUGH_ONE_ROW_RELATION),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java:88]: this rule name is not good. it do not push limit **THROUGH** one row relation

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java [hunk 2]
```diff
@@ -67,65 +74,45 @@ public List<Rule> buildRules() {
...
+                logicalLimit(logicalEmptyRelation())
+                        .then(UnaryNode::child)
+                        .toRule(RuleType.PUSH_LIMIT_THROUGH_EMPTY_RELATION),
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java:91]: ditto, may be eliminate limit on xxxx is better

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java [hunk 3]
```diff
@@ -40,24 +39,32 @@
...
  * Limit can't be push down if it has a valid offset info.
  */
-public class LimitPushDown implements RewriteRuleFactory {
+public class PushdownLimit implements RewriteRuleFactory {
 
     @Override
     public List<Rule> buildRules() {
         return ImmutableList.of(
                 // limit -> join
                 logicalLimit(logicalJoin(any(), any())).whenNot(Limit::hasValidOffset)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java:48]: maybe do not need `join(any(), any())`? `join()` is enough

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java (hunk 2 above)
Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java:91]: we have a file named EliminateLimit

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java (hunk 3 above)
Comment (jackwener) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownLimit.java:48]:
```java
f (newJoin == null || limit.child().children().equals(newJoin.children())) {
                                return null;
                            }
```

use it

**Review by morrySnow — APPROVED** [review]

**Pull request merged**